### PR TITLE
Sign up/sign in tests

### PR DESCRIPTION
**Pull request by i-Gor3040**

### CODE REVIEW

File: cypress.config.js [hunk 1]
```diff
@@ -1,9 +1,13 @@
 const { defineConfig } = require("cypress");
 
 module.exports = defineConfig({  
+  experimentalSessionAndOrigin: true,
```
Comment (DlgSHi) [inline cypress.config.js]: this is no needed parameter in the latest version

File: cypress.config.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+  chromeWebSecurity: true,
```
Comment (DlgSHi) [inline cypress.config.js]: this parameter is set to `true` by default :)

File: cypress.config.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
   viewportWidth: 1920,
   viewportHeight: 1080,
-  e2e: {
+  watchForFileChanges: false,
+    e2e: {
+    excludeSpecPattern: ['**/1-getting-started/*', '**/2-advanced-examples/*'],
```
Comment (DlgSHi) [inline cypress.config.js]: don't exclude them, just delete the files

File: cypress/e2e/landingPage.spec.cy.js [hunk 4]
```diff
@@ -1,27 +1,128 @@
-/// <reference types="cypress" /> // is this really needed?
-import { onLandingPage } from "../support/pageObject/landingPage";
-import { onSignUpPage } from "../support/pageObject/signUpPage";
+import { faker } from "@faker-js/faker";
+// import { onLandingPage } from "../support/pageObject/landingPage";
+// import { onSignUpPage } from "../support/pageObject/signUpPage";
+
+const USERNAME = faker.person.firstName() + faker.number.int({ min: 1, max: 100 });
```
Comment (DlgSHi) [inline cypress/e2e/landingPage.spec.cy.js]: you don't need to add number at the end, `faker` will generate all the time different values

File: cypress/e2e/landingPage.spec.cy.js [hunk 5]
```diff
@@ -1,27 +1,128 @@
...
-       onLandingPage.clickSignUpLink;
-      onSignUpPage.enterUserName;
-      onSignUpPage.enterEmail;
-      onSignUpPage.enterPSWD;
-      onSignUpPage.clickSignUpBtn; 
-      
-      //assert
-    }); 
+  beforeEach("open app", () => {
+    cy.visit("/");
```
Comment (DlgSHi) [inline cypress/e2e/landingPage.spec.cy.js]: you don't want to use POM ?

File: cypress/e2e/landingPage.spec.cy.js [hunk 6]
```diff
@@ -1,27 +1,128 @@
...
+  });
+
+  // Add test with user name already taken
+  it("sign up", () => {
+    cy.contains("Sign up").click();
+    cy.get('[placeholder="Username"]').clear().type(USERNAME);
+
+    cy.get('[placeholder="Email"]')
+      .clear()
+      .type(USERNAME + "@yopmail.com");
```
Comment (DlgSHi) [inline cypress/e2e/landingPage.spec.cy.js]: you can generate same mail with `faker` too by using `faker.internet.email()`

File: cypress/e2e/landingPage.spec.cy.js [hunk 7]
```diff
@@ -1,27 +1,128 @@
...
+
+    cy.get('[placeholder="Password"]')
+      .clear()
+      .type("Test123@");
```
Comment (DlgSHi) [inline cypress/e2e/landingPage.spec.cy.js]: same here with password

File: cypress/e2e/landingPage.spec.cy.js [hunk 8]
```diff
@@ -1,27 +1,128 @@
...
+      });
+
+    cy.get('[show-authed="true"] li').eq(3).click();
+
+    cy.contains("Edit Profile Settings").click();
+
+    cy.contains("button", "Or click here to logout.").click();
   });
 
+  it("login with invalid email", () => {
```
Comment (DlgSHi) [inline cypress/e2e/landingPage.spec.cy.js]: so much duplicated code :)

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: cypress/e2e/landingPage.spec.cy.js (hunk 4 above)
Comment (i-Gor3040) [reply cypress/e2e/landingPage.spec.cy.js]: several times the generated name has already been taken

File: cypress/e2e/landingPage.spec.cy.js (hunk 5 above)
Comment (i-Gor3040) [reply cypress/e2e/landingPage.spec.cy.js]: tried, but it didn't work. Thought we will look into the issue and refactor later

File: cypress/e2e/landingPage.spec.cy.js (hunk 6 above)
Comment (i-Gor3040) [reply cypress/e2e/landingPage.spec.cy.js]: i thought, i might need to receive an email and get some values from there. If not, it's not difficult to introduce changes

File: cypress/e2e/landingPage.spec.cy.js (hunk 8 above)
Comment (i-Gor3040) [reply cypress/e2e/landingPage.spec.cy.js]: still i didn't cover several cases :)

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**